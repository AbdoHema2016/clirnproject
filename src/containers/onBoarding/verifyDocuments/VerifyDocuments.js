import React, {PureComponent} from 'react';
import {
  Text,
  View,
  Platform,
  Keyboard,
  KeyboardAvoidingView,
  TouchableWithoutFeedback,
  Image,
  NativeModules,
  NativeEventEmitter,
  Alert,
  ActivityIndicator,
  SafeAreaView,
} from 'react-native';
import {connect} from 'react-redux';
import {signUpStepAction} from '../personalDetails/redux/actions';
import AsyncStorage, {AsyncConstants} from '../../../utilities/AsyncStorage';
import Style from './style/style';
import CLabel from '../../../components/cLabel';
import {Constants} from '../../../utilities';
import {initiateJumioSDK} from '../../Jumio/SDKMethods';
import {translate} from '../../../Localization';

import CButton from '../../../components/cButton';
import {getJumioCredentials} from './redux/actions';
import CNavigationBackButton from '../../../components/cNavigationBackButton';
import {resetOtpAction} from '../otp/redux/actions';
import {logAnalyticsEvent} from '../../../utilities/Firebase';
import {JumioFlowMethodsObj} from '../../Jumio/verificationFlowMethods';
const {JumioMobileSDKNetverify} = NativeModules;
const {LOCAL_PATH, signUpSteps, testIds, analyticsIds} = Constants;

class VerifyDocuments extends PureComponent {
  componentDidMount() {
    this.setupNetVerifyListeners();
    if (this.props.access_token) {
      this.getJumioCredentials();
    }
    this.navigationOptions();
  }

  navigationOptions = () => {
    this.props.navigation.setOptions({
      title: '',
      headerTransparent: true,
      headerLeft: () => (
        <CNavigationBackButton
          title={translate('BACK.BACK_BUTTON_TITLE')}
          backButtonAction={this.goBack}
        />
      ),
    });
  };

  goBack = () => {
    this.props.navigation.navigate('ContactStack');
    this.props.resetOtpAction();
  };

  getJumioCredentials = () => {
    const {access_token, fetchJumioCredentials} = this.props;
    fetchJumioCredentials({access_token});
  };

  componentWillUnmount() {
    this.emitterNetverify.removeAllListeners('EventDocumentData');
    this.emitterNetverify.removeAllListeners('EventErrorNetverify');
  }
  componentDidUpdate(prevProps) {
    if (this.props.access_token !== prevProps.access_token) {
      this.getJumioCredentials();
    }
  }

  onSkipPress = () => {
    this.props.signUpStepAction('5');
    AsyncStorage.setItemInStorage(AsyncConstants.STEP, signUpSteps.Feeling);
    this.props.navigation.navigate('Feeling');
    logAnalyticsEvent(analyticsIds.jumio_skipped);
  };
  setupNetVerifyListeners = () => {
    this.emitterNetverify = new NativeEventEmitter(JumioMobileSDKNetverify);

    this.emitterNetverify.addListener(
      'EventDocumentData',
      (EventDocumentData) => {
        this.props.signUpStepAction('5');
        AsyncStorage.setItemInStorage(AsyncConstants.STEP, signUpSteps.Feeling);
        this.props.navigation.replace('Feeling');
        this.emitterNetverify.removeListener('EventDocumentData');
        this.emitterNetverify.removeListener('EventErrorNetverify');
      },
    );

    this.emitterNetverify.addListener(
      'EventErrorNetverify',
      (EventErrorNetverify) => {
        this.emitterNetverify.removeListener('EventDocumentData');
        this.emitterNetverify.removeListener('EventErrorNetverify');
        if (EventErrorNetverify.errorCode === 'G00000') {
          return;
        }

        Alert.alert(
          'Error!',
          JumioFlowMethodsObj.getJumioError(EventErrorNetverify.errorMessage),
        );
      },
    );
  };
  openJumioFlow = () => {
    const {apiToken, apiSecret, dataCenter, userId} = this.props;
    logAnalyticsEvent(analyticsIds.jumio_started);
    return initiateJumioSDK({apiToken, apiSecret, dataCenter, userId});
  };
  render() {
    const {loading, error} = this.props;
    if (loading) {
      return (
        <View style={Style.loaderContainer}>
          <ActivityIndicator size={'large'} />
        </View>
      );
    } else if (error) {
      return (
        <SafeAreaView style={Style.errorContainer}>
          <Text style={Style.errorText}>
            {translate('STRINGS.SOMETHING_WENT_WRONG')}
          </Text>
          <CButton
            text={'Retry'}
            buttonCustomStyle={Style.retryButton}
            textStyle={Style.retry}
            onPress={this.getJumioCredentials}
          />
        </SafeAreaView>
      );
    }
    return (
      <View style={Style.container}>
        <KeyboardAvoidingView
          style={Style.keyboardAvoidingView}
          behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
          keyboardVerticalOffset={Platform.OS === 'android' ? 44 : 0}>
          <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
            <View style={Style.innerContainer}>
              <CLabel style={Style.infoStageLabel} text={'4/7'} />
              <CLabel style={Style.pageTitle} text={'ID Verification'} />
              <CLabel
                style={Style.pageSubTitle}
                text={translate('STRINGS.VERIFICATION_MSG')}
              />
              <View style={Style.verifyIconContainer}>
                <Image source={LOCAL_PATH.VERIFY_DOC} />
              </View>
              <View style={Style.footerContainer}>
                <CButton
                  text={'Continue'}
                  textStyle={Style.whiteText}
                  onPress={this.openJumioFlow}
                  buttonContainerStyle={Style.blackButton}
                />
                <Text
                  accessible={true}
                  testID={testIds.skip}
                  accessibilityLabel={testIds.skip}
                  style={Style.skipForNow}
                  onPress={this.onSkipPress}>
                  {translate('STRINGS.SKIP')}
                </Text>
              </View>
            </View>
          </TouchableWithoutFeedback>
        </KeyboardAvoidingView>
      </View>
    );
  }
}

const mapStateToProps = ({
  jumio: {apiSecret, apiToken, dataCenter, loading, error},
  signIn,
  personalDetails: {access_token},
}) => {
  return {
    apiSecret,
    apiToken,
    dataCenter,
    loading,
    error,
    access_token,
    userId: signIn.userID,
  };
};

export default connect(mapStateToProps, {
  fetchJumioCredentials: getJumioCredentials,
  signUpStepAction,
  resetOtpAction,
})(VerifyDocuments);
