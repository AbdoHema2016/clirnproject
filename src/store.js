import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import {rootSaga} from './sagas';
const sagaMiddleware = createSagaMiddleware();
const isDevelopmentEnv =
  process && process.env && process.env.NODE_ENV === 'development';
const isProductionEnv =
  process && process.env && process.env.NODE_ENV === 'production';
const logStoreInfo = false;
const middlewares = [sagaMiddleware];
function actionLogger({getState}) {
  return (next) => (action) => {
    let allowed = [
      'jumio',
      'notificationsReducer',
      'otp',
      'companies',
      'signIn',
      'updateContactDetailsReducer',
      'userProfile',
      'vaccinations',
    ];
    let state = getState();
    let loggedStates = Object.keys(state)
      .filter((key) => allowed.includes(key))
      .reduce((obj, key) => {
        return {
          ...obj,
          [key]: state[key],
        };
      }, {});
    console.info({actionType: action.type, loggedStates});
    const returnValue = next(action);
    return returnValue;
  };
}
if (isDevelopmentEnv && logStoreInfo) {
  middlewares.push(createLogger());
}
if (isProductionEnv) {
  middlewares.push(actionLogger);
}

const store = createStore(rootReducer, applyMiddleware(...middlewares));
sagaMiddleware.run(rootSaga);
export default store;
