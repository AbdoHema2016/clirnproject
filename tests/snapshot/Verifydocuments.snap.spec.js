import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import store from '../../src/store';
import VerifyDocuments from '../../src/containers/onBoarding/verifyDocuments/VerifyDocuments';

jest.mock('@react-navigation/native', () => ({
  withNavigationFocus: (component) => component,
}));

jest.useFakeTimers();

describe('VerifyDocuments screen', () => {
  const props = {
    navigation: {
      setOptions: jest.fn(),
    },
  };
  it('Snapshot VerifyDocuments', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <VerifyDocuments {...props} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
