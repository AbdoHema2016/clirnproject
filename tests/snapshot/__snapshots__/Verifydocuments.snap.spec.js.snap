// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyDocuments screen Snapshot VerifyDocuments 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color={null}
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
